src/nectarchain/dqm/charge_integration.py

```diff
@@ -47,12 +47,14 @@ def ConfigureForRun(self, path, Chan, Samp, Reader1):
 
 
     def ProcessEvent(self, evt, noped):
-        #print("test", evt.r0.tel[0].waveform[0])
         pixel = evt.nectarcam.tel[0].svc.pixel_ids
-        pixel21 = np.arange(0,21,1,dtype = int)
-        pixel = list(pixel)
-        pixel21 = list(pixel21)
-        pixels = np.concatenate([pixel21,pixel])
+        if len(pixel) < self.Chan:
+            pixel21 = np.arange(0,self.Chan - len(pixel),1,dtype = int)
+            pixel = list(pixel)
+            pixel21 = list(pixel21)
+            pixels = np.concatenate([pixel21,pixel])
```

Comment:
I still do not understand why the two lists are concatenated here. In such cases, we end up with 1876 pixels, instead of 1855 with 21 of them being masked out.

Reply:
OK, this was a misunderstanding of the logic on my side.